engine/api/repositories_manager.go

```diff
@@ -171,6 +180,17 @@ func repositoriesManagerOAuthCallbackHandler(w http.ResponseWriter, r *http.Requ
 	cache.Get(cache.Key("reposmanager", "oauth", state), &data)
 	projectKey := data["project_key"]
 	rmName := data["repositories_manager"]
+	username := data["username"]
+
+	u, errU := user.LoadUserWithoutAuth(db, username)
+	if errU != nil {
+		return sdk.WrapError(errU, "repositoriesManagerAuthorizeCallback> Cannot load user %s", username)
+	}
+
+	proj, errP := project.Load(db, projectKey, u)
```

Comment:
No sure it will work. This handler is an oauth callback. I think router will panic here 
if key != "" {
		proj, errproj := project.Load(db, key, c.User, project.LoadOptions.Default)
		if errproj != nil {
			return errproj
		}
		c.Project = proj
	}
because c.User is nil